lib/fluent/plugin/in_forward.rb

```diff
@@ -125,46 +128,30 @@ def configure(conf)
             })
         end
       end
+      @lsock = @usock = nil
     end
 
     def start
       super
 
-      @loop = Coolio::Loop.new
-
       socket_manager_path = ENV['SERVERENGINE_SOCKETMANAGER_PATH']
       if Fluent.windows?
         socket_manager_path = socket_manager_path.to_i
       end
       client = ServerEngine::SocketManager::Client.new(socket_manager_path)
 
       @lsock = listen(client)
-      @loop.attach(@lsock)
+      event_loop_attach(@lsock)
 
       @usock = client.listen_udp(@bind, @port)
       @usock.fcntl(Fcntl::F_SETFL, Fcntl::O_NONBLOCK)
       @hbr = HeartbeatRequestHandler.new(@usock, method(:on_heartbeat_request))
-      @loop.attach(@hbr)
-
-      @thread = Thread.new(&method(:run))
+      event_loop_attach(@hbr)
     end
 
-    def shutdown
-      # In test cases it occasionally appeared that when detaching a watcher, another watcher is also detached.
-      # In the case in the iteration of watchers, a watcher that has been already detached is intended to be detached
-      # and therfore RuntimeError occurs saying that it is not attached to a loop.
-      # It occurs only when testing for sending responses to ForwardOutput.
-      # Sending responses needs to write the socket that is previously used only to read
-      # and a handler has 2 watchers that is used to read and to write.
-      # This problem occurs possibly because those watchers are thought to be related to each other
-      # and when detaching one of them the other is also detached for some reasons.
-      # As a workaround, check if watchers are attached before detaching them.
-      @loop.watchers.each {|w| w.detach if w.attached? }
-      @loop.stop
-      @usock.close
-      @thread.join
-      @lsock.close
-
+    def stop
```

Comment:
`stop` is called before `shutdown`, right?
What happens when close socket before shutdown/close event loop?
Is it safe?